Caprica/common/CapricaReportingContext.h

```diff
@@ -42,24 +42,31 @@ struct CapricaReportingContext final {
   void exitIfErrors();
 
   template <typename... Args>
-  NEVER_INLINE void error(CapricaFileLocation location, fmt::format_string<Args...> msg, Args&&... args) {
+  ALWAYS_INLINE void warning(CapricaFileLocation location, size_t warningNumber, std::string_view msg, Args&&... args) {
```

Comment:
This shift means we no longer have the ability to get compile-time checking of format strings for warnings and errors :( Is it really worth losing that just to shift to std::format?

Reply:
I didn't work on it for too long, but switching over to `std::format_string<Args...>` resulted in a compile time error I couldn't quite figure out from the macros. I would say that in the short term it's not really a big deal, and it results in one less third-party dependency. Regardless I'll look into it some more.

Reply:
Actually thinking on it more using string_view and passing that to `vformat` as I did here does at least do compile time argument checking, but not compile time format checking because that's not a thing until c++26.